### PR TITLE
feat: Mark Unmarked Attendance

### PR DESCRIPTION
**Pull request by Anurag810**
This feature allow the user to mark unmarked attendance for a month. 
Marked_unmarked_attendance

### CODE REVIEW

File: erpnext/hr/doctype/attendance/attendance.py [hunk 1]
```diff
@@ -89,17 +90,85 @@ def add_attendance(events, start, end, conditions=None):
...
+	today = get_datetime()
+
+	dates_of_month = ['{}-{}-{}'.format(today.year, month_map[month], r) for r in range(1, calendar.monthrange(today.year, month_map[month])[1] + 1)]
+
+	length = len(dates_of_month)
+	month_start, month_end = dates_of_month[0], dates_of_month[length-1]
+
+
+	records = frappe.get_all("Attendance", fields = ['attendance_date', 'employee'] , filters = [
+		["attendance_date", ">", "2019-12-1"],
```
Comment (nabinhait) [inline erpnext/hr/doctype/attendance/attendance.py]: hard-coded dates

File: erpnext/hr/doctype/attendance/attendance.py [hunk 2]
```diff
@@ -89,17 +90,85 @@ def add_attendance(events, start, end, conditions=None):
...
+			'status': data.status,
+			'company': company,
+		}
+		attendance = frappe.get_doc(doc_dict).insert()
+		attendance.submit()
+
+
+def get_month_map():
+	return frappe._dict({
+		"January":1,
```
Comment (nabinhait) [inline erpnext/hr/doctype/attendance/attendance.py]: space after :

File: erpnext/hr/doctype/attendance/attendance.py [hunk 3]
```diff
@@ -89,17 +90,85 @@ def add_attendance(events, start, end, conditions=None):
...
+
+
+	records = frappe.get_all("Attendance", fields = ['attendance_date', 'employee'] , filters = [
+		["attendance_date", ">", "2019-12-1"],
+		["attendance_date", "<", "2019-12-31"],
+		["employee", "=", employee],
+		["docstatus", "!=", 2]
+	])
+
+	marked_day = [get_datetime(record.attendance_date) for record in records]
```
Comment (nabinhait) [inline erpnext/hr/doctype/attendance/attendance.py]: marked_days and unmarked_days

**Review by nabinhait — APPROVED** [review]

**Comment by somfinso** [comment]
image
I am using erpnext 13.i am trying to using **Mark Attendance** but geeting this erro when i click on mark attendance.
suggestion and help will highly appcreciated

Thank you

**Pull request merged**